aiopg/connection.py

```diff
@@ -421,3 +428,24 @@ def __del__(self):
     def notifies(self):
         """Return notification queue."""
         return self._notifies
+
+
+class _CursorContextManager:
+    def __init__(self, conn, impl, timeout, echo):
+        self._conn = conn
+        self._impl = impl
+        self._timeout = timeout
+        self._echo = echo
+
+    def __iter__(self):
+        impl = yield from self._impl
+        return Cursor(self._conn, impl, self._timeout, self._echo)
+
+    @asyncio.coroutine
+    def __aenter__(self):
+        self._cursor = yield from self
+        return self._cursor
+
+    @asyncio.coroutine
+    def __aexit__(self, type, value, traceback):
+        self._cursor.close()
```

Comment:
I don't think that's necessary.  For __aexit__ to be called by the with statement, __aenter__ must have been successfully executed, so the cursor should exist.